de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetRoleManagement.java

```diff
@@ -130,10 +147,17 @@ protected void createTableColumns(EditingDomain editingDomain) {
 			@Override
 			public void mouseDown(MouseEvent e) {
 				if (tableViewer.getTable().getColumnCount() > 1 && e.x > tableViewer.getTable().getColumn(0).getWidth()) {
-					openCustomDialog();
+					// Check if the user has the required permission before opening the custom dialog
+		            if (hasPermission()) {

```

Comment:
I think it would be better to add this implementation into the isActive / isEnabled method of the editing domain....